packages/backend/src/open_payments/account/model.ts

```diff
@@ -25,4 +27,64 @@ export class Account
 
   public readonly assetId!: string
   public asset!: Asset
+
+  // The proportion of the account's total received amount for which
+  // there are `account.web_monetization` webhook events.
+  // The value should be equivalent to the following query:
+  // select sum(`withdrawalAmount`) from `webhookEvents` where `withdrawalAccountId` = `account.id`
+  public totalEventsAmount!: bigint
+  public processAt!: Date | null
+
+  public async onCredit({
+    totalReceived,
+    withdrawalCron
+  }: OnCreditOptions): Promise<Account> {
+    if (this.asset.withdrawalThreshold !== null) {
+      const account = await Account.query()
+        .patchAndFetchById(this.id, {
+          processAt: new Date()
+        })
+        .whereRaw('?? <= ?', [
+          'totalEventsAmount',
+          totalReceived - this.asset.withdrawalThreshold
+        ])
+        .withGraphFetched('asset')
+      if (account) {
+        return account
+      }
+    }
+    if (withdrawalCron && !this.processAt) {
+      await this.$query().patch({
+        processAt: parser.parseExpression(withdrawalCron).next().toDate()
```

Comment:
I was picturing the wallet being able to configure regularly scheduled withdrawals like every day at 5pm EST.
Maybe `withdrawalThrottleDelay` could be added in addition to cron (would they need to be mutually exclusive?)

Reply:
I think we should try to minimize the handling logic for this in Rafiki, since it is so wallet-specific. Regularly scheduled withdrawals are really better suited to a pull/query model (i.e. the wallet could query all accounts with money to withdraw), rather than webhooks/events.

What about a hybrid? i.e. use the `processAt: new Date(Date.now() + config.withdrawalThrottleDelay`, but allow the wallet operator to disable starting any `processNextAccount` workers when they configure Rafiki. Expose an endpoint that, when called, queries accounts with `now < processAt` and triggers `n` of them (for some integer parameter `n`).

That gives a bit more control to the wallet operator. It's less elegant than a pull model would be, but it reuses more logic from the push model.